prefect_shell/commands.py

```diff
@@ -88,6 +93,9 @@ def example_shell_run_command_flow():
                     f"Command failed with exit code {process.returncode}:\n" f"{stderr}"
                 )
                 raise RuntimeError(msg)
+        
+        if os.path.exists(tmp.name):
+            os.remove(tmp.name)
```

Comment:
I think we should wrap this in a `try/finally` statement so that if something errors in the middle, it'll still get erased.